admin-dashboard/admin.js

```diff
@@ -0,0 +1,202 @@
+                requestTitle.innerHTML = request.title;
+                row.appendChild(requestTitle);
+
+                // create the column to add the request description and append to the row
+                let requestDescription = document.createElement('td');
+                requestDescription.innerHTML = request.description;
+                row.appendChild(requestDescription);
+
+                // create the column to add the request date and append to the row
+                let requestDate = document.createElement('td');
+            let dateRequested = new Date(request.date_requested);
+            requestDate.innerHTML = dateRequested.toDateString();
+                row.appendChild(requestDate);
+
+                // create the column to add the request status and append to the row
+                let requestStatus = document.createElement('td');
+                requestStatus.innerHTML = request.status;
+                row.appendChild(requestStatus);
+
+                // create the column to add the buttons for responding to requests and append to the row
+                let requestActions = document.createElement('td');
+                if (request.status === 'Pending Approval') {
+                    // if the request is pending approval, then two buttons are added:
+                    // approve button
+                    let approveButton = document.createElement('button');
+                    approveButton.className = 'btn btn-success';
+                    approveButton.innerHTML = 'Approve';
+                    approveButton.onclick = respondRequest;
+                    requestActions.appendChild(approveButton);
+
+                    // disapprove button
+                    let disapproveButton = document.createElement('button');
+                    disapproveButton.className = 'btn btn-danger';
+                    disapproveButton.innerHTML = 'Disapprove';
+                    disapproveButton.onclick = respondRequest;
+                    requestActions.appendChild(disapproveButton);
+                } else if (request.status === 'Approved') {
+                    // if the request is approved, then we add the resolve button
+                    let resolveButton = document.createElement('button');
+                    resolveButton.className = 'btn btn-neutral';
+                    resolveButton.innerHTML = 'Resolve';
+                    resolveButton.onclick = respondRequest;
+                    requestActions.appendChild(resolveButton);
+                } else if (request.status === 'Disapproved' || request.status === 'Resolved') {
+                    // if the request is disapproved or resolved then no actions are possible
+                    requestActions.innerHTML = '-'
+                }
+                row.appendChild(requestActions);
+                // add this row to the requests container
+                requestsContainer.appendChild(row);
+            }
+        );
+    }
+}
```

Comment:
The above approach is great but I find it hard to read. You can use es6 template literals to create your HTML markup.
For example.
```
const table = `
<table style="width:100%">
  <tr>
    <th>Firstname</th>
    <th>Lastname</th> 
    <th>Age</th>
  </tr>
  <tr>
    <td>Jill</td>
    <td>Smith</td> 
    <td>50</td>
  </tr>
  <tr>
    <td>Eve</td>
    <td>Jackson</td> 
    <td>94</td>
  </tr>
</table>
`;
```